src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -986,7 +986,8 @@ void primitive_inst::realloc_if_needed(bool prev_execution_skipped) {
             // TODO: check need_reset_output_memory per output
             if (need_reset_output_memory() && !can_be_optimized()) {
                 GPU_DEBUG_TRACE_DETAIL << id() << " : Need reset output memory considering user" << std::endl;
-                add_dep_event(_outputs[i]->fill(get_network().get_stream()));
+                auto dep_event = get_network().get_stream().enqueue_marker(_impl_params->dep_events);
+                add_dep_event(_outputs[i]->fill(get_network().get_stream(), {dep_event}));
```

Comment:
Could you explain why you used `marker` instead of adding `dep_events` directly?

Reply:
oh I see. then could you add a comment there about that? I think this would be difficult to understand without prior knowledge.